### ISSUE TITLE
Documentation should be more precise about allowed array shapes

### ISSUE THREAD

**Issue by billmark** [deepmind/optax#205]
optax.softmax_cross_entropy accepts inputs of shape [..., num_classes] and returns a result of shape [...].  This behavior should be clearly documented but is not.

Likewise for other loss functions.

**Comment by mtthss** [comment]
Fixed in https://github.com/deepmind/optax/pull/210
